### DISCUSSION TITLE
Inconsitent speed

### DISCUSSION THREAD

**Discussion by oguztudo** [srsran/srsRAN_Project#106] [General Help] [answered]
Hey everybody, when i use the gNB with my phone i get results that vary a loooot. 
With iperf i never manage to get past 50Mbits. Somehow using an internet speed test will result in higher Numbers. I even managed to go past 100Mbits. This number can drop to 40Mbits in the next test. I testetd with regular wifi and with my pc connected via ethernet and both give me constant results. I also got to mention that my pc's internet connection is way past 500Mbits for UL and DL so this couldnt be a bottleneck. As for cpu usage: i am not even close to use the processors on full power.
The performance script is activated though.

Maybe some of you encountered the same issue and could provide help to to get constant results.

**Comment by gustavobsch** [comment]
I have similar inconsistencies, been wondering why it does that. I have a b210 and one test I reach 72mbit and then the next one 37mb. I think the problem may be related to using my laptop as gNB because I have other apps open and they might be generating interrupts, seems like gnb its very sensitive to cpu interrupts

**Reply by oguztudo** [reply]
I truly cannot think that my cpu is struggling to be honest. 
Even when i set the BW to 100MHz my cpu is only using two cores at 50% on average, there rest is almost sleepinh. I get massive lates and Over/Underflows tho on 100MHz but thats probably a whole other story.
Also i am almost using stock kubuntu with just the necessary programs for srsran installed...
Did you monitor your cpu usage to see if thats a bottleneck?


I would be really glad if theres a fix to make thinks work with conistency

**Reply by gustavobsch** [reply]
I did monitor the CPU and it wasn't more than 10%, it wasn't about CPU type, speed or usage. 
You might want to read #10 there's some good info there. 

What SDR are you using BTW? to be able to reach 100Mhz BW

**Reply by oguztudo** [reply]
The info is good but i did not find it helpful in my case because running lower than 100MHz i have barely Overflow issues. It was just an extreme example.

I am running a USRP X310 :)

**Reply by herlesupreeth** [reply]
Speedtest over internet can vary a lot based on the server it chooses and load on that server. Best way to test is run iPerf between UE and Core Network.

**Reply by oguztudo** [reply]
By Core network you mean just running iperf  on a console in the host machine?

If so i did that but my results were slower than the achievable max with speed tests...

**Reply by herlesupreeth** [reply]
By Core Network I mean the following

I am assuming a setup with open5gs
 
1. For testing DL throughput, running a iPerf server in UPF and iPerf client on the phone
2. For testing DL throughput, running a iPerf client in UPF and iPerf server on the phone

**Reply by oguztudo** [reply]
Thanks for the info!
Maybe im asking the obvious here but how am i able to start iperf in the upf?

**Reply by herlesupreeth** [reply, marked as answer]
if you want to start the server its `iperf -s -u -i 1`

For `iperf client iperf -u -c <IPERF_SERVER_IP> -t <TIME_TO_RUN_IPERF_TEST> -b <MAX_RATE_TO_TEST> -i 1`

e.g. (assuming iPerf server is running on a machine with IP 192.168.100.2, test for 20 seconds, 150Mbps max rate)

```
iperf -s -u -i 1
iperf -u -c 192.168.100.2 -t 20 -b 150M -i 1
```

**Reply by oguztudo** [reply]
Thanks for the quick reply!

Oh i that is what i was doing. I thought there are extra steps necessary to start the iperf inside of the upf (for example an extra command the iperf to upf). As far as i know the main traffic goes through the upf anyway.

**Reply by herlesupreeth** [reply]
Ah I see. So you are saying with iPerf there was inconsistent throughput??

**Reply by ismagom** [reply]
To achieve best speed, try first with iperf UDP. then in the console type `t` and you will see the trace printed. That's useful to debug. Can you paste the output of that here? Additionally, logs in info level are also useful
